src/compiler/transformers/test/rewrite-aliased-paths.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { CompilerCtx } from '@stencil/core/declarations';
+import { mockCompilerCtx, mockValidatedConfig } from '@stencil/core/testing';
+import path from 'path';
+import ts from 'typescript';
+
+import { rewriteAliasedDTSImportPaths, rewriteAliasedSourceFileImportPaths } from '../rewrite-aliased-paths';
+import { transpileModule } from './transpile';
+
+async function pathTransformTranspile(component: string) {
+  const compilerContext: CompilerCtx = mockCompilerCtx();
+  const config = mockValidatedConfig();
+
+  const mockPathsConfig: ts.CompilerOptions = {
+    paths: {
+      '@namespace': [path.join(config.rootDir, 'name/space.ts')],
+      '@namespace/subdir': [path.join(config.rootDir, 'name/space/subdir.ts')],
+    },
+    declaration: true,
+  };
+
+  // we need to have files in the `inMemoryFs` which TypeScript
+  // can resolve, otherwise it won't find the module and won't
+  // transform the module ID
+  await compilerContext.fs.writeFile(path.join(config.rootDir, 'name/space.ts'), 'export const foo = x => x');
+  await compilerContext.fs.writeFile(path.join(config.rootDir, 'name/space/subdir.ts'), 'export const bar = x => x;');
```

Comment:
in order for these tests to work TypeScript needs to be able to resolve the files we're importing from, so here in this little helper function we instantiate an in-memory FS and write two little modules to it. TypeScript is patched to read from the in-memory FS, so when we try to resolve modules in our transformer it will see these.